pkg/controllers/apps/sidekick_controller.go

```diff
@@ -81,6 +83,28 @@ func (r *SidekickReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	if sidekick.DeletionTimestamp != nil {
+		if core_util.HasFinalizer(sidekick.ObjectMeta, appsv1alpha1.SchemeGroupVersion.Group) {
+			if err := r.terminate(ctx, &sidekick); err != nil {
+				return ctrl.Result{}, err
+			}
+
+			return ctrl.Result{}, nil
```

Comment:
May be just `return ctrl.Result{}, r.terminate(ctx, &sidekick)` ?

Reply:
Also, Can you resolve the linter errors ?

Reply:
most linter errors are fixed by https://github.com/kubeops/sidekick/pull/5 . I guess we can fix the other ones there too.